app/scripts/services/hpa.js

```diff
@@ -99,6 +102,13 @@ angular.module("openshiftConsole")
     };
 
 
+    var hasV2HPAAnnotations = function(hpaResources) {
+      return _.some(hpaResources, function(hpa) {
+        return JSON.parse(annotation(hpa, 'autoscaling.alpha.kubernetes.io/metrics'));
```

Comment:
Missed this before (sorry), but if the annotation is not valid JSON -- for instance, if edited by the user -- it will be a runtime error. We should add a try/catch here if we need JSON, but I would just skip the parse since we don't do anything with the value.

Reply:
Hmm. will the save be successful if the user edits but the JSON is invalid?  Prob my bad assuming we would reject the change...

Reply:
Either way we don't want the runtime error in the client that stops JS execution.